### PR TITLE
#7 Find the largest palindrome made from the product of two 3-digit numbers.

### PR DESCRIPTION
**Pull request by sanishmaharjan**
Please code review.

### CODE REVIEW

File: src/com/lftechnology/java/training/sanish/palindrome/Main.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+/**
+ * Class Main
+ * 
+ * @author Sanish Maharjan <sanishmaharjan@lftechnology.com>
+ */
+public class Main {
+	public static void main(String args[]){
+		final Logger LOGGER = Logger.getLogger(Main.class.getName());
+		try {
+			LOGGER.info("Largest palindrome made from the product of two 3-digit numbers: "+ Palindrome.getLargestPalindrome(3));
```
Comment (bibhushanJoshi) [inline src/com/lftechnology/java/training/sanish/palindrome/Main.java]: better to use parameterized logger.
http://stackoverflow.com/questions/17716381/slf4j-parameterized-logging-using-varargs-method

File: src/com/lftechnology/java/training/sanish/palindrome/Palindrome.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+/**
+ * Provide palindrome calculation functions
+ * @author Sanish Maharjan <sanishmaharjan@lftechnology.com>
+ */
+public class Palindrome {
+	/**
+	 * Calculate and return largest palindrome number made from the product of two provided digit numbers
+	 * 
+	 * @param digitLength digit length of two product from which have find largest palindrome number
```
Comment (bibhushanJoshi) [inline src/com/lftechnology/java/training/sanish/palindrome/Palindrome.java]: should define datatype of param as well @param digitLength {@link Integer}

File: src/com/lftechnology/java/training/sanish/palindrome/Palindrome.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+	 */
+	public static int getLargestPalindrome(int digitLength) throws Exception {
+		final Logger LOGGER = Logger.getLogger(Main.class.getName());
+		int maxNum = (int) Math.pow(10, digitLength) - 1;
+		int minNum = (int) Math.pow(10, digitLength - 1);
+		if (digitLength <= 4) {
+			for (int i = maxNum; i > minNum; i--) {
+				for (int j = maxNum; j > minNum; j--) {
+					int multipleNum = j * i;
+					if (isPalindromNum(multipleNum)) {
```
Comment (bibhushanJoshi) [inline src/com/lftechnology/java/training/sanish/palindrome/Palindrome.java]: mutipleNum should be intialized outside the loop. Cause the value will be initialized every time the loop run.

File: src/com/lftechnology/java/training/sanish/palindrome/Palindrome.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+	 *            {@link Integer} number that have to check is palindrom number
+	 * @return {@link Boolean} true if supplied number is palindrom else false
+	 * @author Sanish Maharjan <sanishmaharjan@lftechnology.com>
+	 */
+	public static boolean isPalindromNum(long number) {
+		long preNumber = number;
+		long reverse = 0;
+		while (number != 0) {
+			reverse = reverse * 10;
+			reverse = reverse + number % 10;
```
Comment (nimesh-mishra) [inline src/com/lftechnology/java/training/sanish/palindrome/Palindrome.java]: use reverse+=number%10; It is shorter.

File: src/com/lftechnology/java/training/sanish/palindrome/Palindrome.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+	 * Check is supplied number is palindrom number or not
+	 * 
+	 * @param number
+	 *            {@link Integer} number that have to check is palindrom number
+	 * @return {@link Boolean} true if supplied number is palindrom else false
+	 * @author Sanish Maharjan <sanishmaharjan@lftechnology.com>
+	 */
+	public static boolean isPalindromNum(long number) {
+		long preNumber = number;
+		long reverse = 0;
```
Comment (bhuwang) [inline src/com/lftechnology/java/training/sanish/palindrome/Palindrome.java:52]: you can simply use StringBuilder.reverse() method

File: src/com/lftechnology/java/training/sanish/palindrome/Palindrome.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+	 * @param digitLength
+	 *            {@link Integer} digit length of two product from which have
+	 *            find largest palindrome number
+	 * @return {@link Integer} largest palindrome number
+	 * @throws Exception
+	 *             Digit length greater than 4 not supported
+	 * @author Sanish Maharjan <sanishmaharjan@lftechnology.com>
+	 */
+	public static long getLargestPalindrome(int digitLength) throws Exception {
+		final Logger LOGGER = Logger.getLogger(Main.class.getName());
```
Comment (bhuwang) [inline src/com/lftechnology/java/training/sanish/palindrome/Palindrome.java]: Logger must be a class variable and it access modifier should be private.

**Pull request merged**